features/permanent_slots.feature

```diff
@@ -0,0 +1,34 @@
+Feature: permanent slots
+  Scenario: check that physical permanent slots are created
+    Given I start postgres0
+    Then postgres0 is a leader after 10 seconds
+    And there is a non empty initialize key in DCS after 15 seconds
+    When I issue a PATCH request to http://127.0.0.1:8008/config with {"slots": {"test_physical": {"type": "physical"}}, "postgresql": {"parameters": {"wal_level": "logical"}}}
+    Then I receive a response code 200
+    And Response on GET http://127.0.0.1:8008/config contains slots after 10 seconds
+    Then postgres0 has a physical replication slot named test_physical after 10 seconds
+    And I start postgres1
+
+  @slot-advance
+  Scenario: check that logical permanent slots are created
+    Given I run patronictl.py restart batman postgres0 --force
+    And I issue a PATCH request to http://127.0.0.1:8008/config with {"slots": {"test_logical": {"type": "logical", "database": "postgres", "plugin": "test_decoding"}}}
+    Then postgres0 has a logical replication slot named test_logical with the test_decoding plugin after 10 seconds
+
+  @slot-advance
+  Scenario: check that permanent slots are created on the replica
+    Given postgres1 has a logical replication slot named test_logical with the test_decoding plugin after 10 seconds
+    Then Logical slot test_logical is in sync between postgres0 and postgres1 after 10 seconds
+    And postgres1 has a physical replication slot named test_physical after 2 seconds
```

Comment:
Is the creation of the physical slot directly related to the existence of the logical slot in the `Given` statement? I feel like it would exist as a result of starting postgres1? If that is right should there then be a `Scenario` between the first and second that has `Given I start postgres1` with this `postgres1 has a physical replication slot named test_physical`?

Reply:
Thats totally true, but we need to start postgres1 in the first scenario, because otherwise the last scenario will fail on 9.6 and 10 (they don't run scenarios tagged with `@slot-advance`).

Reply:
Should we keep claiming to support Postgres versions in Patroni that are not supported by PGDG?
IMO that kind of incentives people that are running old versions to keep running them.
I'd say it would be a bit less relaxed if we support the 5 versions supported by PGDG + the one that just went EOL, so people can move out in the meantime.
What are your thoughts on that?

Reply:
I see, ok! TBH I wouldn't expect people to be running on 9.3 e.g. -- although there are always weird cases 😆 .

Reply:
~i can hear greenplum users crying~